src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/finalparameters/InputFinalParametersRecordForLoopPatternVariables.java

```diff
@@ -0,0 +1,20 @@
+/*
+FinalParameters
+ignorePrimitiveTypes = (default)false
+ignoreUnnamedParameters = (default)true
+tokens = PATTERN_VARIABLE_DEF,FOR_EACH_CLAUSE
+
+
+*/
+
+// non-compiled with javac: Compilable with Java20
+package com.puppycrawl.tools.checkstyle.checks.finalparameters;
+
+public class InputFinalParametersRecordForLoopPatternVariables {
+    record ARecord(String name, int age) {
+    }
+    static void method(final ARecord[] records) {
+        for (ARecord(String name, final int age) : records) { // violation, 'name' should be final
```

Comment:
please add "same" code below that is compliant  by Check, so we will see clearly what is bad and what is expected.

please apply the same rule to all other test code your created.